Add :hideall: tag for liquid_tags.include_code plugin

alberand opened the pull request:
There is possibility to hide filename but there is no possibility to hide all elements (title + download link)
The following code:
```
{% include_code buffering/cat1.c lang:c :hideall: %}
```
Will produce something like this:
![image](https://user-images.githubusercontent.com/8555162/64723267-80297f00-d4d0-11e9-8140-1e9c1e61ed21.png)

Lucas-C commented:
Nice !

First, could you include the plugin name as part of this PR name ?

The tests for this plugin will fixed by https://github.com/getpelican/pelican-plugins/pull/1178
Could you maybe add a unit test for this new feature please ?

alberand commented:
So, currently there is no tests for luqid_tags.**code_include** at all, right?

Lucas-C commented:
You mean `include_code` ? No, there doesn't seem to be any test covering it right now

alberand commented:
I see that there is mix of pytest's tests (most of them) and unittest tests (test_notebook.py) I suppose it is better if I stick with pytest?

Lucas-C commented:
_cf._ https://github.com/getpelican/pelican-plugins/issues/1170

We are currently trying to have all `nostests` based unit tests to pass in our CI.
We plan to use `pytest` once Travis CI is green.
So it's up to you basically :)

alberand commented:
I added tests for regex and `include_code()`; all passes. Also, I fixed one problem with `codec` parameter (see [edcfdf6](https://github.com/getpelican/pelican-plugins/pull/1194/commits/edcfdf687f78f81ae92e75973375ae9a8ba539c0)). 

I am not sure if it possible to create dummy preprocessor to use it in the tests, do you know something about it? For now I just created empty object with required functions (see lines `60 - 72` in `test_include_code.py`)

Lucas-C commented:
Thanks for the tests and the fix !
Could you please just rebase your branch on this repo `master` branch ?

alberand commented:
Done.

justinmayer commented:
@krischer / @talha131: Any feedback on this pull request before it is merged?

talha131 commented:
@justinmayer will get to you by Sunday.

talha131 commented:
@justinmayer to the best of my knowledge PR is fine.

@alberand great addition. Useful!

alberand commented:
Merge?😁

justinmayer commented:
Many thanks to @alberand for the contribution (replete with tests!) and to @Lucas-C and @talha131 for reviewing. 👏
